packages/algob/src/internal/deployer.ts

```diff
@@ -147,6 +151,36 @@ class DeployerBasicMode {
     }
   }
 
+  /**
+   * Returns SSC checkpoint key using application index,
+   * returns undefined if it doesn't exist
+   * @param index Application index
+   */
+  getSSCCPKeyFromId (index: number): string | undefined {
+    const resultMap = this.cpData.precedingCP[this.networkName]?.ssc ?? new Map();
+    for (const [key, nestedMap] of resultMap) {
+      if ([...nestedMap][nestedMap.size - 1][1].appID === index) {
```

Comment:
we want to get the last object? As far as I know, the order is not predictable in JS, so not sure if the thing above will work. Also there is no need to construct an array.

Reply:
I did research on this, I found that we cannot directly get last value from a map, the way to do it is to traverse the whole map and choose the last element.
Could you please tell me the other way ?

Reply:
What is the last element? What order are you going to use? If it's min / max then you can do it using one loop.

Reply:
the last element have the latest timestamp

Reply:
Ok, got it.
 - traverse the map store value whose key is maximum